### ISSUE TITLE
AttributeRouteStrategy ignores ODataRoutePrefix when generating the path template

### ISSUE THREAD

**Issue by jalstrup** [rbeauchamp/Swashbuckle.OData#71]
We are running into an issue with the path templates not being generated correctly when using an ODataRoutePrefix. Our OData endpoint is set up in the following way. 

**Config**
```c#
config.MapODataServiceRoute("AdminOData", "api/v1/admin", ODataEdmModelProvider.GetEdmModel());
```

**Controller**

```c#
    [ODataRoutePrefix("products")]
    public class ProductsController : ODataController
    {
        [EnableQuery]
        [ODataRoute("({key})/attributevalues({attributevalueKey})")]
        public IHttpActionResult GetAttributeValue(Guid key, Guid attributevalueKey)
        {
            return this.Ok();
        }
    }
```

It looks like the ODataActionDescriptor is created by the AttributeRouteStrategy. To generate the path template it looks like it is using the ODataRoute prefix (in this scenario "api/v1/admin") and  ODataRouteAttribute.PathTemplate (in this scenario "({key})/attributevalues({attributevalueKey})") to generate the path template. The resulting path template ends up looking like the following. 

api/v1/admin/({key})/attributevalues({attributevalueKey})

I would expect it to be 

api/v1/admin/products({key})/attributevalues({attributevalueKey})

I think the same logic that is being used by AttributeRoutingConvention.GetODataPathTemplate would be needed to generate the correct path.

**Comment by rbeauchamp** [comment]
Hey @jalstrup, thanks for posting this issue. I'll look into it.

**Comment by mjolka** [comment]
@rbeauchamp I've created [a branch](https://github.com/mjolka/Swashbuckle.OData/tree/issue_71) that hopefully fixes this problem. Please let me know if you have any feedback, or would be willing to merge.

**Comment by rbeauchamp** [comment]
@mjolka Thank you! Please submit a pull request and I'll check it out.

**Comment by rbeauchamp-lendingtree** [comment]
Fixed by #84.

**Comment by rbeauchamp-lendingtree** [comment]
Fix is available in v2.18.0
